src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -559,28 +591,104 @@ Namespace Microsoft.CodeAnalysis.Semantics
+                    operandLazy = New Lazy(Of IOperation)(Function() CreateBoundDelegateCreationExpressionChildOperation(DirectCast(operand, BoundDelegateCreationExpression)))
+                Else
+                    ' This is an error scenario in which we have a delegate conversion, but it failed to bind correctly, so the child here
+                    ' is either a bad expression or a method group. Delegate to standard operation handling for this.
+                    Debug.Assert(operand.Kind = BoundKind.MethodGroup OrElse
+                                 operand.Kind = BoundKind.BadExpression OrElse
+                                 operand.Kind = BoundKind.AddressOfOperator)
+                    operandLazy = New Lazy(Of IOperation)(Function() Create(operand))
+                End If
+                Return (methodSymbol, operandLazy, isAddressOfDelegateCreation:=True)
+            Else
+                Dim methodSymbol As MethodSymbol = Nothing
+                Return (methodSymbol, New Lazy(Of IOperation)(Function() Create(operand)), isAddressOfDelegateCreation:=False)
+            End If
+        End Function
+
+        Private Function CreateBoundDelegateCreationExpressionOperation(boundDelegateCreationExpression As BoundDelegateCreationExpression) As IDelegateCreationExpression
+            Dim syntax As SyntaxNode = boundDelegateCreationExpression.Syntax
+            Dim type As ITypeSymbol = boundDelegateCreationExpression.Type
+            Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundDelegateCreationExpression.ConstantValueOpt)
+            Dim isImplicit As Boolean = boundDelegateCreationExpression.WasCompilerGenerated OrElse boundDelegateCreationExpression.Syntax.Kind() = SyntaxKind.AddressOfExpression
+
+            Dim target As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() CreateBoundDelegateCreationExpressionChildOperation(boundDelegateCreationExpression))
+
+            Return New LazyDelegateCreationExpression(target, _semanticModel, syntax, type, constantValue, isImplicit)
+        End Function
+
+        Private Function CreateBoundDelegateCreationExpressionChildOperation(boundDelegateCreationExpression As BoundDelegateCreationExpression) As IMethodReferenceExpression
+            Dim method As IMethodSymbol = boundDelegateCreationExpression.Method
+            Dim isVirtual As Boolean = method IsNot Nothing AndAlso
+                                               (method.IsAbstract OrElse method.IsOverride OrElse method.IsVirtual) AndAlso
+                                               Not boundDelegateCreationExpression.SuppressVirtualCalls
+            Dim instance As Lazy(Of IOperation) =
+                        New Lazy(Of IOperation)(Function() Create(If(boundDelegateCreationExpression.ReceiverOpt, boundDelegateCreationExpression.MethodGroupOpt?.ReceiverOpt)))
+            Dim syntax As SyntaxNode = boundDelegateCreationExpression.Syntax
```

Comment:
>Dim syntax As SyntaxNode = boundDelegateCreationExpression.Syntax [](start = 12, length = 65)

It is not clear why this is the right syntax to use for the method reference. For example, why not use syntax of the MethodGroupOpt, when on is available?

Reply:
This is probably something about strange bound tree shape around AddressOf. Comments would be very helpful if this is the case.

---
In reply to: [142810723](https://github.com/dotnet/roslyn/pull/22056#discussion_r142810723) [](ancestors = 142810723)